pkg/testutils/centralgrpc/connect_to_central.go

```diff
@@ -73,14 +73,20 @@ func UnauthenticatedGRPCConnectionToCentral(t *testing.T) *grpc.ClientConn {
 }
 
 // GRPCConnectionToCentral returns a GRPC connection to Central, which can be used in E2E tests.
-// It fatals the test if there's an error.
+// Basic auth is used to establish the connection. It fatals the test if there's an error.
 func GRPCConnectionToCentral(t testutils.T) *grpc.ClientConn {
 	return grpcConnectionToCentral(t, func(opts *clientconn.Options) {
 		opts.ConfigureBasicAuth(RoxUsername(t), RoxPassword(t))
 	})
 }
 
-func grpcConnectionToCentral(t testutils.T, optsModifyFunc func(options *clientconn.Options)) *grpc.ClientConn {
+// GRPCConnectionToCentralWithOpts returns a GRPC connection to Central, which can be used in E2E tests.
+// Use optsFuncs to configure the connection (ie: to use basic auth, token auth, handle TLS, etc.)
+func GRPCConnectionToCentralWithOpts(t testutils.T, optsFuncs ...func(opts *clientconn.Options)) *grpc.ClientConn {
```

Comment:
The previous iteration of this change modified the original method, `GRPCConnectionToCentral`, to add opts just as you described - [after review](https://github.com/stackrox/stackrox/pull/12568#discussion_r1760658834) the thought was that this may cause confusion because, the way I had it, when any option was set (perhaps unrelated to auth) then auth would break.

Initially it seemed there wasn't a clean way to detect if an auth option was provided, re-thinking this I should be able to pre-iterate through all the options, and then check if auth was modified, and if not set basic auth - all within the original `GRPCConnectionToCentral` method - not the cleanest but for testing purposes should do just fine.